### PR TITLE
Add first version of utils collection for object oriented programming

### PR DESCRIPTION
**Pull request by lincrosenbach**

### CODE REVIEW

**Comment by kfasolin** [comment]
Did you move all foundation folder? I thought we would move only interface and the decorators for now

**Comment by lincrosenbach** [comment]
> Did you move all foundation folder? I thought we would move only interface and the decorators for now

Had to move the dependencies for interfaces and indirect dependencies resulting on those files.

**Comment by kfasolin** [comment]
Another doubt, we will use this as a package? Or in development mode?

**Comment by nicoddemus** [comment]
> Another doubt, we will use this as a package? Or in development mode? And did you already have a branch with the changes on ben10?

I believe as a package... the code here rarely changed in the past years.

File: appveyor.yml [hunk 1]
```diff
@@ -8,9 +8,6 @@ environment:
     - PYTHON: "C:\\Python36"
       TOXENV: "docs"
 
-    - PYTHON: "C:\\Python27"
```
Comment (nicoddemus) [inline appveyor.yml]: We should update our cookiecutter to remove this dependency

File: setup.py [hunk 2]
```diff
@@ -30,7 +29,7 @@
     license="MIT license",
     long_description=readme + "\n\n" + history,
     include_package_data=True,
-    python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*",
```
Comment (nicoddemus) [inline setup.py]: Ditto about the cookiecutter

Comment (nicoddemus) [reply setup.py]: Done it in https://github.com/ESSS/cookiecutter-esss-pypackage/pull/5 👍

File: appveyor.yml (hunk 1 above)
Comment (lincrosenbach) [reply appveyor.yml]: :+1:

**Comment by lincrosenbach** [comment]
> The `.pydevproject` is intended to be committed?

Dont know for sure.

**Comment by nicoddemus** [comment]
I think it is supposed to live in the repository

**Pull request merged**